clipper_admin/clipper_manager.py

```diff
@@ -578,13 +584,119 @@ def _save_python_function(self, name, predict_function):
                 "See http://clipper.ai/documentation/python_model_deployment/ for more information."
             )
 
+        # Export modules used by predict_function not captured in anaconda or pip
+        self._export_local_modules(
+            c.modules, serialization_dir, conda_dep_fname, pip_dep_fname,
+            local_modules_folder_name, local_modules_list_fname)
+        print("Supplied local modules")
+
         # Write out function serialization
         func_file_path = os.path.join(serialization_dir, predict_fname)
         with open(func_file_path, "w") as serialized_function_file:
             serialized_function_file.write(serialized_prediction_function)
         print("Serialized and supplied predict function")
         return serialization_dir
 
+    def _get_already_exported_modules(self, serialization_dir,
+                                      conda_deps_fname, pip_deps_fname):
+        """
+        Returns a list of names of modules that will be installed on the container.
+        This list includes the names of exported conda and pip modules as well as
+        those of preinstalled modules.
+
+        Returns
+        -------
+        list
+            List of names of modules whose installations are already accounted for
+        """
+        conda_deps_abs_path = os.path.join(serialization_dir, conda_deps_fname)
+        pip_deps_abs_path = os.path.join(serialization_dir, pip_deps_fname)
+        python_container_conda_deps_fname = os.path.join(
+            cur_dir, "python_container_conda_deps.txt")
+
+        get_module_name = lambda line: line.strip().split('=')[0]
+
+        ignore_list = []
+
+        if (os.path.isfile(conda_deps_abs_path)):
+            with open(conda_deps_abs_path, 'r') as f:
+                for line in f:
+                    ignore_list.append(get_module_name(line))
+
+        if (os.path.isfile(pip_deps_fname)):
+            with open(pip_deps_abs_path, 'r') as f:
+                for line in f:
+                    ignore_list.append(get_module_name(line))
+
+        with open(python_container_conda_deps_fname, 'r') as f:
+            for line in f:
+                ignore_list.append(line.strip())
+
+        return ignore_list
+
+    def _export_local_modules(self, func_modules, serialization_dir,
+                              conda_deps_fname, pip_deps_fname,
+                              local_modules_folder_name,
+                              local_modules_list_fname):
+        """
+        Supplies modules in `func_modules` not already accounted for by Anaconda or pip to container.
+        Copies those modules to a location specified by `serialization_dir` and `local_modules_folder_name`
+        Enumerates the local modules copied in a file whose location is specified by `serialization_dir` and
+        `local_module_deps_fname`.
```

Comment:
By "location is specified by serialization_dir and local_module_deps_fname" you mean it will be at `serialization_dir/local_module_deps_fname`?